nodejs/eks/nodegroup.ts

```diff
@@ -1685,3 +1668,71 @@ function createManagedNodeGroupInternal(
 
     return nodeGroup;
 }
+
+/**
+ * getRecommendedAMI returns the recommended AMI to use for a given configuration
+ * when none is provided by the user.
+ *
+ * See: https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html
+ */
+function getRecommendedAMI(
+    args: Omit<NodeGroupOptions, "cluster"> | Omit<NodeGroupV2Options, "cluster">,
+    k8sVersion: pulumi.Output<string>,
+    parent: pulumi.Resource | undefined,
+): pulumi.Input<string> {
+    const amiType = getAMIType(args.amiType, args.gpu, args.instanceType);
+    const amiID = k8sVersion.apply((v) => {
+        const parameterName = `/aws/service/eks/optimized-ami/${v}/${amiType}/recommended/image_id`;
+        return pulumi.output(aws.ssm.getParameter({ name: parameterName }, { parent, async: true }))
+            .value;
+    });
+
+    return amiID;
+}
+
+/**
+ * isGravitonInstance returns true if the instance type is a Graviton instance.
+ * We determine this by checking if the third character of the instance type is
+ * a "g".
+ *
+ * See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html.
+ */
+function isGravitonInstance(instanceType: pulumi.Input<string> | undefined): boolean {
+    if (!instanceType) {
```

Comment:
I think technically the first part can be multiple characters - like `Hpc7g`.

Reply:
Ah! Yes you're right. I'll need to account multi-char instance families.

Reply:
Updated to use regex.